src/components/TodoItem.tsx

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+/* eslint-disable */
+
+import React, { useState } from 'react';
+import { Todo } from '../types/Todo';
+import cn from 'classnames';
+
+type Props = {
+  todo: Todo;
+  isLoading?: boolean;
+  onDeleteTodo: (todoId: number) => Promise<void>;
+  onUpdateTodo: (todo: Todo) => Promise<void>;
+  isInEditMode?: boolean;
+  setEditedTodoId: (id: number | null) => void;
+};
+
+export const TodoItem: React.FC<Props> = props => {
+  const {
+    todo,
+    onDeleteTodo,
+    onUpdateTodo,
+    isInEditMode,
+    isLoading,
+    setEditedTodoId,
+  } = props;
+
+  const [todoTitleValue, setTodoTitleValue] = useState(todo.title);
+
+  const inputRef = React.useRef<HTMLInputElement>(null);
+
+  const onCheckTodo = () => {
+    const todoToUpdate = { ...todo, completed: !todo.completed };
+
+    onUpdateTodo(todoToUpdate);
+  };
+
+  const onDoubleClick = () => {
+    setEditedTodoId(todo.id);
+  };
+
+  const onBlur = async (event: React.FocusEvent<HTMLFormElement, Element> | React.FocusEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const normalizedTitle = todoTitleValue.trim();
+
+    if (todo.title === normalizedTitle) {
+      setEditedTodoId(null);
+
+      return;
+    }
+
+    try {
+      if (normalizedTitle === '') {
```

Comment:
use this syntax for all the cases(more readable)

```suggestion
      if (!normalizedTitle.length) {
```